### PR TITLE
Use sublocality when locality is missing - AddressSearch

### PR DESCRIPTION
**Pull request by aldo-expensify**
### Details
There are addresses that have the component "locality" missing. We use the "locality" as the city. This PR will make us use the "sublocality" as the city when the "locality" is missing.

### Fixed Issues

$ https://github.com/Expensify/App/issues/5867

### Tests / QA Steps

1. Create account in NewDot, verify it and log in
2. Create Workspace
3. Start add VBA flow and get to the "Company information" step (/bank-account/company)
4. Input in the "Company address" field: `64 Noll Street, Brooklyn, NY, USA` and select the first (and only) result
5. Fill the rest of the fields
6. You should be able to continue to the next step without errors.

### Tested On

- [x] Web
- [ ] Mobile Web
- [ ] Desktop
- [ ] iOS
- [ ] Android

### CODE REVIEW

**Review by roryabraham — CHANGES_REQUESTED** [review]
> Do console.debug logging eventually reach our logging servers?

Nope, please use the [`Log` library](https://github.com/Expensify/App/blob/main/src/libs/Log.js). See [this PR](https://github.com/Expensify/App/pull/5827).

**Comment by aldo-expensify** [comment]
> Nope, please use the [`Log` library](https://github.com/Expensify/App/blob/main/src/libs/Log.js). See [this PR](https://github.com/Expensify/App/pull/5827).

Thanks, I would think these logs are interesting enough to be sent to the server. They may eventually reveal other problematic cases that we don't know about, sounds correct?

**Comment by aldo-expensify** [comment]
Update: Changed `console.debug` for `Log.hmmm`

File: src/components/AddressSearch.js [hunk 1]
```diff
@@ -67,7 +81,11 @@ const AddressSearch = (props) => {
             // Gather the values from the Google details
             const streetNumber = getAddressComponent(details, 'street_number', 'long_name');
             const streetName = getAddressComponent(details, 'route', 'long_name');
-            const city = getAddressComponent(details, 'locality', 'long_name');
+            let city = getAddressComponent(details, 'locality', 'long_name');
+            if (!city) {
+                city = getAddressComponent(details, 'sublocality', 'long_name');
+                Log.hmmm('Replacing missing locality with sublocality: ', {address: details.formatted_address, sublocality: city});
```
Comment (roryabraham) [inline src/components/AddressSearch.js:87]: I would add the `[AddressSearch]` prefix here as well.

**Review by roryabraham — COMMENTED** [review]
Log changes look good. Let's add some unit tests for this!

**Comment by aldo-expensify** [comment]
> Log changes look good. Let's add some unit tests for this!

Thanks for the review, I'll be looking into the testing stuff tomorrow to finish this up!

File: tests/unit/GooglePlacesUtilsTest.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+                        short_name: 'US',
+                        types: ['country', 'political'],
+                    },
+                    {
+                        long_name: '92027',
+                        short_name: '92027',
+                        types: ['postal_code'],
+                    },
+                ],
+                formatted_address: 'Quail Ridge Rd, Escondido, CA 92027, USA',
```
Comment (aldo-expensify) [inline tests/unit/GooglePlacesUtilsTest.js:40]: I left this here even if it not used in the tests just as a way to remember these addresses that gave up problems.

**Comment by aldo-expensify** [comment]
hmm, any ideas on why this could be happening?

**Comment by TomatoToaster** [comment]
Seems unrelated to your changes, @aldo-expensify, try merging the main branch into your PR, it might fix that issue.

**Review by roryabraham — CHANGES_REQUESTED** [review]
Thanks for adding these tests, that's huge. Just had some minor suggestions.

**Review by roryabraham — CHANGES_REQUESTED** [review]
Sorry, one last thing 😅

**Review by roryabraham — APPROVED** [review]

**Pull request merged**